Gems/Atom/RHI/Vulkan/Code/Source/RHI/SwapChain.cpp

```diff
@@ -117,8 +130,12 @@ namespace AZ
                 auto& presentationQueue = device.GetCommandQueueContext().GetOrCreatePresentationCommandQueue(*this);
                 m_presentationQueue = &presentationQueue;
 
-                result = CreateSwapchain();
-                RETURN_RESULT_IF_UNSUCCESSFUL(result);
+                if (IsDefaultSwapChainNeeded())
```

Comment:
I thought we added the check during pipeline creation time so why do we need this check again here. It should never come here if the default pipeline is never created for Android?

Reply:
It's not creating the default render pipeline (so it'll eventually won't render to the default swapchain), but the swapchains are always created by the WindowsContext, it's a different initialization path. After this change now it's doing both, not creating the default swapchain and not creating the default render pipeline.